### PR TITLE
Add max_member_weight to sum_of_members selector semantics (and change to oneof).

### PR DESCRIPTION
**Pull request by jonathan-dilorenzo**
As per #396. I found the phrasing a bit awkward. Maybe y'all have better ideas?

### CODE REVIEW

File: docs/v1/P4Runtime-Spec.mdk [hunk 1]
```diff
@@ -1599,25 +1599,33 @@ The `ActionProfile` message includes the following fields:
   the `selector_size_semantics` value as described below.
 
 * `max_group_size`, an `int32` which is 0 for an Action Profile, or, for an
-  Action Selector, its semantics is specified by the
-  `selector_size_semantics` value as described below.
+  Action Selector, its semantics is specified by the  `selector_size_semantics`
```
Comment (antoninbas) [inline docs/v1/P4Runtime-Spec.mdk:1602]: nit: you have 2 spaces before `selector_size_semantics`

File: docs/v1/P4Runtime-Spec.mdk [hunk 2]
```diff
@@ -1599,25 +1599,33 @@ The `ActionProfile` message includes the following fields:
...
   specifies how `size` and `max_group_size` are interpreted. It can be either:
-    * `SUM_OF_WEIGHTS`, indicating that `size` and `max_group_size` represent
+    * `sum_of_weights`, indicating that `size` and `max_group_size` represent
       the maximum sum of weights that can be present across all selector groups
       and within a single selector group respectively.
-    * `SUM_OF_MEMBERS`, indicating that `size` and `max_group_size` represent
+    * `sum_of_members`, indicating that `size` and `max_group_size` represent
       the maximum number of members that can be present across all selector
-      groups and within a single selector group respectively.
+      groups and within a single selector group respectively. The `SumOfMembers`
```
Comment (antoninbas) [inline docs/v1/P4Runtime-Spec.mdk:1615]:
```suggestion
      groups and within a single selector group respectively, irrespective of their weight. The `SumOfMembers`
```
not introduced by this PR, but I think it's good to emphasize

File: docs/v1/P4Runtime-Spec.mdk [hunk 3]
```diff
@@ -1599,25 +1599,33 @@ The `ActionProfile` message includes the following fields:
...
+      message used to represent this value also contains an optional int32
+      `max_member_weight`, which indicates the maximum weight of each
+      individual member. If unset, any 32-bit integer is allowable for weight.
```
Comment (antoninbas) [inline docs/v1/P4Runtime-Spec.mdk:1618]: I'd say "allowed" sounds better than "allowable" here, but that may be a matter of preference

File: docs/v1/P4Runtime-Spec.mdk [hunk 4]
```diff
@@ -3610,10 +3618,11 @@ An `ActionProfileGroup` entity update message has the following fields:
       existing port on the device, otherwise the server must return
       `INVALID_ARGUMENT`.
 
-* `max_size` is the maximum sum of all member weights for the group. This field
-  is defined when the group is inserted, and must not be changed in a
-  `MODIFY` update. See the subsection below for the [rules on setting
-  `max_size`](#sec-max-size-rules).
+* `max_size` is the maximum sum of all members or member weights (as per the
```
Comment (antoninbas) [inline docs/v1/P4Runtime-Spec.mdk:3621]: I guess this is not specific to this PR, but we forgot to make that change in the previous PR?

**Review by antoninbas — COMMENTED** [review]
only a couple of nits, otherwise LGTM

File: docs/v1/P4Runtime-Spec.mdk (hunk 4 above)
Comment (jonathan-dilorenzo) [reply docs/v1/P4Runtime-Spec.mdk:3621]: Yes, that's right. The INVALID_ARGUMENT_ERROR part was implied below, but I got some internal comments of confusion so I thought I'd clarify at the same time.

**Comment by jonathan-dilorenzo** [comment]
Fixed! Thanks for comments!

**Comment by antoninbas** [comment]
@jonathan-dilorenzo will you need help with generated files?

**Comment by jonathan-dilorenzo** [comment]
Yes please! I gave it another shot, but I have some permission issue when I try it :'(.

**Comment by antoninbas** [comment]
@jonathan-dilorenzo done, I updated your branch

File: docs/v1/P4Runtime-Spec.mdk [hunk 5]
```diff
@@ -3610,10 +3619,11 @@ An `ActionProfileGroup` entity update message has the following fields:
...
+  `selector_size_semantics`) for the group. This field is defined when the group
+  is inserted, and must not be changed in a  `MODIFY` update, otherwise an
```
Comment (antoninbas) [inline docs/v1/P4Runtime-Spec.mdk:3624]: nit: I spotted another double whitespace ...

Comment (jonathan-dilorenzo) [reply docs/v1/P4Runtime-Spec.mdk]: Nooooo! Mistakes were made. I even searched to make sure this was the last spot.

**Review by antoninbas — APPROVED** [review]

**Comment by antoninbas** [comment]
@smolkaj any comment on this? otherwise I'll go ahead and merge it

**Review by smolkaj — APPROVED** [review]

**Comment by smolkaj** [comment]
Looks good to me.

**Pull request merged**